File: source-base/sep.backend.v1/Data/Entities/BusEnrollment.cs
namespace sep.backend.v1.Data.Entities
{
    public class BusEnrollment : BaseEntity
    {
      
        public int Id { get; set; }
        public int BusId { get; set; }
        public int? PupilId { get; set; }
        public int? BusSupervisorId { get; set; }
        public int SemesterId { get; set; }
        public int? BusStopId { get; set; }
        public virtual Semester? Semester { get; set; }
        public virtual Bus? Bus { get; set; }
        public virtual BusSupervisor? BusSupervisor { get; set; }
        public virtual Pupil? Pupil { get; set; }
        public virtual BusStop? BusStop { get; set; }

    }
}

File: source-base/source.base.fe/src/types/model/bus-enrollment.ts
export interface ResponseBusEnrollmentData {
  pageNumber: number;
  pageSize: number;
  firstPage: string;
  lastPage: string;
  totalPages: number;
  totalRecords: number;
  nextPage: string | null;
  previousPage: string | null;
  data: BusEnrollment[];
}

export interface BusEnrollment {
  id: number;
  busId: number;
  pupilId: number;
  busSupervisorId: number | null;
  semesterId: number;
  pupilName: string;
  pupilCode: string;
  semesterName: string;
  academicYear: string;
  busName: string;
  busSupervisorName: string | null;
  busStopId: number;
  busStopName: string | null;
}

export interface RequestBusEnrollmentIndex {
  pageNumber: number | null;
  pageSize: number | null;
  busId?: number | null;
  semesterId?: number | null;
}

export interface CreateBusEnrollment {
  busId: number;
  pupilId: number | null;
  semesterId: number;
  busSupervisorId: number | null;
  busStopId: number | null;
}

export interface PupilDTOResponse {
  id: number;
  firstName: string;
  lastName: string;
  email: string;
  gender: boolean;
  genderName: string;
  donorName: string;
  donorPhoneNumber: string;
  address: string;
  dateOfBirth: string;
  schoolId: number;
  schoolName: string | null;
  accountStatus: number;
  accountStatusName: string;
  username: string;
  image: string; 
}

export interface BusSupervisorDTOResponse {
  id: number;
  firstName: string;
  lastName: string;
  phoneNumber: string;
  address: string;
  gender: boolean;
  schoolId: number;
}
